pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -189,9 +197,22 @@ func (n *nodeLocalDNS) WaitCleanup(ctx context.Context) error {
 }
 
 func (n *nodeLocalDNS) computeResourcesData() (map[string][]byte, error) {
-	var (
-		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+	healthAddress := ""
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		n.values.IPVSAddress = nodelocaldnsconstants.IPVSAddress
+		healthAddress = nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		n.values.IPVSAddress = nodelocaldnsconstants.IPVSIPv6Address
+		healthAddress = fmt.Sprintf("[%s]", nodelocaldnsconstants.IPVSIPv6Address)
```

Comment:
Is it possible to always use the style of `[2001:db8::1]` for IPv6 to unify the handling?

Reply:
From what I've read you only use the brackets in combination with a port. Or is this more common?

Reply:
I've tested it, unfortunately it does not work.

Reply:
```
2024/10/21 06:44:06 [FATAL] Error parsing flags - invalid localip specified - "[fd30:1319:f1e:230b::1]", Exiting
```